### PR TITLE
Lookup fields implementation

### PR DESCRIPTION
**Pull request by gosiexon-zen**
## Description
This code add support for Lookup fields for custom objects in the Copenhagen Theme v4.
[Jira: Lookup fields in the request form or templating API v4](https://zendesk.atlassian.net/browse/GG-3449)
## Screenshots

https://github.com/user-attachments/assets/8d04e0ea-a095-448c-bbd9-47ff0e848f55

## Checklist

- [ ] :green_book: all commit messages follow the [conventional commits](https://conventionalcommits.org/) standard
- [ ] :arrow_left: changes are compatible with RTL direction
- [ ] :wheelchair: Changes to the UI are [tested for accessibility](./../README.md#accessibility-testing) and compliant with [WCAG 2.1](https://www.w3.org/TR/WCAG21/).
- [ ] :memo: changes are tested in Chrome, Firefox, Safari and Edge
- [ ] :iphone: changes are responsive and tested in mobile
- [ ] :+1: PR is approved by @zendesk/vikings

### CODE REVIEW

**Review by luis-almeida — APPROVED** [review]

File: src/modules/new-request-form/fields/LookupField.tsx [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+          const { custom_object_record } = await res.json();
+          const newSelectedOption = {
+            name: custom_object_record.name,
+            value: custom_object_record.id,
+          };
+          setSelectedOption(newSelectedOption);
+          setInputValue(custom_object_record.name);
+        }
+      } catch (error) {
+        console.log(error);
```
Comment (BrunoBFerreira) [inline src/modules/new-request-form/fields/LookupField.tsx:93]: Do we usually leave `console.log` calls in the code for frontend?

**Review by kubraokcu — APPROVED** [review]

File: src/modules/new-request-form/fields/LookupField.tsx (hunk 1 above)
Comment (gosiexon-zen) [reply src/modules/new-request-form/fields/LookupField.tsx:93]: Usually linters don't allow this, here linter tells me that every path needs to return value so I need to do something with this error and we don't have any logging tool added. I will change it to console.error, I see there is other usage of this in Copenhagen Theme repo.

Comment (BrunoBFerreira) [reply src/modules/new-request-form/fields/LookupField.tsx:93]: Thank you for checking this out. If `console.log` is the best approach for it I believe it is fine to keep. Just wanted to be sure how we usually do it.

**Review by BrunoBFerreira — APPROVED** [review]

**Pull request merged**